zcash_keys/src/keys.rs

```diff
@@ -450,6 +450,9 @@ pub struct UnifiedAddressRequest {
 }
 
 impl UnifiedAddressRequest {
+    /// Construct a new unified address request from its constituent parts.
+    ///
+    /// Returns `None` if the resulting unified address would not include at least one shielded receiver.
```

Comment:
I realise this is just documenting the current behaviour, but it should be creating an R1 UA if there are only transparent receivers.

Reply:
The revision of the UA is not tracked at this level; that's an encoding concern for `zcash_address`